app/src/main/java/teamagam/ConnectionTest/connectionTestActivity.java

```diff
@@ -0,0 +1,80 @@
+package teamagam.ConnectionTest;
+
+import android.os.Bundle;
+import android.support.v4.content.ContextCompat;
+import android.support.v7.app.AppCompatActivity;
+import android.view.View;
+import android.widget.EditText;
+import android.widget.TextView;
+
+public class ConnectionTestActivity extends AppCompatActivity {
+
+    private EditText mIpEditText;
+    private TextView mHttpResultTextView;
+    private TextView mPingResultTextView;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_connection_test);
+        setupUiElements();
+        testConnection();
+    }
+
+    public class DisplayResultToTextView implements ValidatorAsyncTask.Displayer {
+        private TextView mResultTextView;
+
+        public DisplayResultToTextView(TextView resultTextView) {
+            mResultTextView = resultTextView;
+            displayLoading();
+        }
+
+        private void displayLoading() {
+            mResultTextView.setText(R.string.checking);
+            mResultTextView.setTextColor(ContextCompat.getColor(ConnectionTestActivity.this, R.color.blue));
+        }
+
+        @Override
+        public void displayOK() {
+            mResultTextView.setText(R.string.positive_result);
+            mResultTextView.setTextColor(ContextCompat.getColor(ConnectionTestActivity.this, R.color.green));
+        }
+
+        @Override
+        public void displayFaild() {
+            mResultTextView.setText(R.string.no_answer_result);
+            mResultTextView.setTextColor(ContextCompat.getColor(ConnectionTestActivity.this, R.color.red));
```

Comment:
This is the third time you call the two methods above together. Extract a method!
private void displayStatus(String status, int color);